### ISSUE TITLE
Major Frames issues after a while flying around

### ISSUE THREAD

**Issue by YumatanGames** [Fabulously-Optimized/fabulously-optimized#356]
### Describe the bug

I'm not sure why, but I'm having major FPS and TPS drops, both in single and multiplayer, after flying around for a while. I did some testing with a completely vanilla client and it doesn't happen. I'm not sure what is happening, as the log doesn't show anything weird as the game is running, although it gives some erros when closing the world and then crashes the client. 

Here are some screenshots I took while flying around. It starts perfectly, with over 500fps sometimes, and then it starts going down. Eventually, it becomes unplayable until I close the client and open again. 

I did notice this before, but I though was some combination of other mods I installed later. Today I decide to do the testing without any mods besides what is included in the modpack and the problem persists. 

Please, let me know if there's any other information I can provide to help you figure it out! :)

![2022-06-01_21 03 08](https://user-images.githubusercontent.com/80188726/171521209-1e1c863d-136f-44d8-97c3-7618fa2b4d00.png)
![2022-06-01_20 30 20](https://user-images.githubusercontent.com/80188726/171521216-f79d0457-54f3-45a8-b2d6-b62c61a515fe.png)
![2022-06-01_20 30 42](https://user-images.githubusercontent.com/80188726/171521220-4a6fe3ae-b7b7-40ee-9fa0-24f45688d84a.png)
![2022-06-01_20 58 36](https://user-images.githubusercontent.com/80188726/171521224-8d2cf7b5-366c-45d8-8219-8a9139b12cb3.png)
![2022-06-01_21 02 53](https://user-images.githubusercontent.com/80188726/171521226-c8c96ad0-0b01-4848-b4f8-af80ed72405a.png)
![2022-06-01_21 03 04](https://user-images.githubusercontent.com/80188726/171521229-8e8d8643-b5a6-4b98-a998-1a3ed7c497b7.png)

### Modpack version

3.8.3 for 1.18.2 (MultiMC)

### Launcher

MultiMC

### Other things

- [ ] I updated FO to a newer version (mention, what version to what version)
- [ ] I removed or added some mods (mention, which ones)

### Additional context

<details>

```
[20:57:25] [Render thread/WARN]: Ambiguity between arguments [teleport, location] and [teleport, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[20:57:25] [Render thread/WARN]: Ambiguity between arguments [teleport, location] and [teleport, targets] with inputs: [0.1 -0.5 .9, 0 0 0]
[20:57:25] [Render thread/WARN]: Ambiguity between arguments [teleport, destination] and [teleport, targets] with inputs: [Player, 0123, @e, dd12be42-52a9-4a91-a8a1-11c01849e498]
[20:57:25] [Render thread/WARN]: Ambiguity between arguments [teleport, targets] and [teleport, destination] with inputs: [Player, 0123, dd12be42-52a9-4a91-a8a1-11c01849e498]
[20:57:25] [Render thread/WARN]: Ambiguity between arguments [teleport, targets, location] and [teleport, targets, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[20:57:25] [Render thread/INFO]: Loaded 7 recipes
[20:57:25] [Render thread/INFO]: Loaded 1141 advancements
[20:57:25] [Render thread/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[20:57:25] [Render thread/INFO]: Applied 0 biome modifications to 0 of 61 new biomes in 247.9 ?s
[20:57:25] [Server thread/INFO]: Starting integrated minecraft server version 1.18.2
[20:57:25] [
[2022-06-01-1.log.gz](https://github.com/Fabulously-Optimized/fabulously-optimized/files/8819501/2022-06-01-1.log.gz)
[latest.log](https://github.com/Fabulously-Optimized/fabulously-optimized/files/8819502/latest.log)
Server thread/INFO]: Generating keypair
[20:57:29] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[20:57:29] [Render thread/INFO]: Preparing spawn area: 0%
[20:57:30] [Render thread/INFO]: Preparing spawn area: 0%
[20:57:30] [Render thread/INFO]: Preparing spawn area: 3%
[20:57:31] [Render thread/INFO]: Preparing spawn area: 8%
[20:57:31] [Render thread/INFO]: Preparing spawn area: 12%
[20:57:32] [Render thread/INFO]: Preparing spawn area: 18%
[20:57:32] [Render thread/INFO]: Preparing spawn area: 23%
[20:57:33] [Render thread/INFO]: Preparing spawn area: 30%
[20:57:33] [Render thread/INFO]: Preparing spawn area: 35%
[20:57:34] [Render thread/INFO]: Preparing spawn area: 36%
[20:57:34] [Render thread/INFO]: Preparing spawn area: 42%
[20:57:35] [Render thread/INFO]: Preparing spawn area: 47%
[20:57:35] [Render thread/INFO]: Preparing spawn area: 54%
[20:57:36] [Render thread/INFO]: Preparing spawn area: 60%
[20:57:36] [Render thread/INFO]: Preparing spawn area: 68%
[20:57:37] [Render thread/INFO]: Preparing spawn area: 73%
[20:57:37] [Render thread/INFO]: Preparing spawn area: 79%
[20:57:38] [Render thread/INFO]: Preparing spawn area: 86%
[20:57:38] [Render thread/INFO]: Preparing spawn area: 92%
[20:57:38] [Server thread/INFO]: Changing view distance to 20, from 10
[20:57:38] [Server thread/INFO]: Changing simulation distance to 6, from 0
[20:57:38] [Render thread/INFO]: Time elapsed: 9360 ms
[20:57:38] [Server thread/INFO]: Yumatan[local:E:336653c0] logged in with entity id 809 at (-17.5, 80.0, 15.5)
[20:57:38] [Server thread/INFO]: Yumatan joined the game
[20:57:39] [Render thread/INFO]: Started 10 worker threads
[20:57:39] [Render thread/INFO]: Loaded 0 advancements
[20:57:43] [Server thread/INFO]: [Yumatan: Set own game mode to Spectator Mode]
[20:57:43] [Render thread/INFO]: [CHAT] Set own game mode to Spectator Mode
[20:58:13] [Server thread/INFO]: Saving and pausing game...
[20:58:13] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:overworld
[20:58:13] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_end
[20:58:13] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_nether
[20:58:20] [Render thread/INFO]: Flushed changes to Minecraft configuration
[20:58:25] [Render thread/WARN]: Reclaimed 3264 bytes at address 2170457855568 that were leaked from an unknown location (logging is disabled)
[20:58:25] [Render thread/WARN]: Reclaimed 21760 bytes at address 2170422994944 that were leaked from an unknown location (logging is disabled)
[20:58:25] [Render thread/WARN]: Reclaimed 8892 bytes at address 2170528763056 that were leaked from an unknown location (logging is disabled)
[20:58:25] [Render thread/WARN]: Reclaimed 59280 bytes at address 2170550194224 that were leaked from an unknown location (logging is disabled)
[20:58:25] [Render thread/WARN]: Reclaimed 432 bytes at address 2169823202464 that were leaked from an unknown location (logging is disabled)
[20:58:25] [Render thread/WARN]: Reclaimed 2880 bytes at address 2169900501040 that were leaked from an unknown location (logging is disabled)
[20:58:25] [Render thread/WARN]: Reclaimed 6048 bytes at address 2169825403456 that were leaked from an unknown location (logging is disabled)
[20:58:25] [Render thread/WARN]: Reclaimed 20960 bytes at address 2170452984016 that were leaked from an unknown location (logging is disabled)
[20:58:36] [Render thread/INFO]: [CHAT] Saved screenshot as 2022-06-01_20.58.36.png
[21:00:36] [Server thread/INFO]: Saving and pausing game...
[21:00:36] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:overworld
[21:00:36] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_end
[21:00:36] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_nether
[21:00:38] [Server thread/INFO]: [Yumatan: Set own game mode to Creative Mode]
[21:00:38] [Render thread/INFO]: [CHAT] Set own game mode to Creative Mode
[21:00:43] [Server thread/INFO]: Yumatan has reached the goal [Sky's the Limit]
[21:00:43] [Render thread/INFO]: [CHAT] Yumatan has reached the goal [Sky's the Limit]
[21:00:43] [Render thread/INFO]: Loaded 5 advancements
[21:01:07] [Render thread/WARN]: Reclaimed 2544 bytes at address 2170450651664 that were leaked from an unknown location (logging is disabled)
[21:01:07] [Render thread/WARN]: Reclaimed 134880 bytes at address 2170429752864 that were leaked from an unknown location (logging is disabled)
[21:01:07] [Render thread/WARN]: Reclaimed 864 bytes at address 2169854471632 that were leaked from an unknown location (logging is disabled)
[21:01:07] [Render thread/WARN]: Reclaimed 5760 bytes at address 2169733547088 that were leaked from an unknown location (logging is disabled)
[21:01:07] [Render thread/WARN]: Reclaimed 20232 bytes at address 2170538064336 that were leaked from an unknown location (logging is disabled)
[21:01:07] [Render thread/WARN]: Reclaimed 16960 bytes at address 2169862699312 that were leaked from an unknown location (logging is disabled)
[21:01:48] [Server thread/INFO]: [Yumatan: Set own game mode to Spectator Mode]
[21:01:48] [Render thread/INFO]: [CHAT] Set own game mode to Spectator Mode
[21:01:59] [Server thread/INFO]: [Yumatan: Set own game mode to Creative Mode]
[21:01:59] [Render thread/INFO]: [CHAT] Set own game mode to Creative Mode
[21:02:53] [Render thread/INFO]: [CHAT] Saved screenshot as 2022-06-01_21.02.53.png
[21:03:04] [Render thread/INFO]: [CHAT] Saved screenshot as 2022-06-01_21.03.04.png
[21:03:09] [Render thread/INFO]: [CHAT] Saved screenshot as 2022-06-01_21.03.08.png
[21:03:23] [Server thread/INFO]: Saving and pausing game...
[21:03:23] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:overworld
[21:03:23] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_end
[21:03:23] [Server thread/INFO]: Saving chunks for level 'ServerLevel[New World]'/minecraft:the_nether
[21:03:24] [Server thread/INFO]: Yumatan lost connection: Disconnected
[21:03:24] [Server thread/INFO]: Yumatan left the game
[21:03:24] [Server thread/INFO]: Stopping singleplayer server as player logged out
[21:03:24] [Server thread/INFO]: Stopping server
[21:03:24] [Server thread/INFO]: Saving players
[21:03:24] [Server thread/INFO]: Saving worlds

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Server thread"

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Thread-1"

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "IO-Worker-18"

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Worker-Main-13"

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Render thread"

Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Worker-Main-3"
AL lib: (EE) alc_cleanup: 1 device not closed
Process exited with code 1.
Clipboard copy at: 01 Jun 2022 21:07:19 -0300
```

</details>

Edit by Madis0: formatted and collapsed logs

**Comment by Madis0** [comment]
> Reclaimed 3264 bytes at address 2170457855568 that were leaked from an unknown location (logging is disabled)

These lines are interesting, though I'm not sure what they really refer to.
Could you disable the JVM flags that are added by this modpack in instance settings? You could also try disabling the RAM limits.

**Comment by YumatanGames** [comment]
> These lines are interesting, though I'm not sure what they really refer to. Could you disable the JVM flags that are added by this modpack in instance settings? You could also try disabling the RAM limits.

I'm not sure how to do this... Would you mind helping me? I'm using MultiMC launcher, where should I go to change those configs?

**Comment by YumatanGames** [comment]
Ok, I'm not sure if it that what you meant, but I disabled Java Arguments and Memory on the Settings tab in MultiMC (I just unticked the boxes) and it made it better. I still have some spikes in TPS and FPS but they are not happening that often and are sparse, not compromising gameplay at all. I'll do some more testing later today. Thanks for the insight.

**Comment by Madis0** [comment]
Yes, that is what I meant. When you test today, please confirm whether you actually needed to uncheck both boxes or just one of them (I am assuming memory is the culprit).

**Comment by Madis0** [comment]
Got it, then I'll probably remove both in the next version.

**Comment by Julienraptor01** [comment]
Aren't the JVMs flags in FO Just the vanilla ones ?

**Comment by Madis0** [comment]
> Aren't the JVMs flags in FO Just the vanilla ones ?

They are from the Fabric Loader installer -> vanilla launcher. But I guess there may still be differences in launchers which cause this, e.g. LWGJL and Java versions.

**Comment by isXander** [comment]
Isn't that a memory leak from some mod?

**Comment by isXander** [comment]
CaffeineMC/sodium-fabric#1024 - I wouldn't touch memory settings and just wait for this to be fixed.

**Comment by Madis0** [comment]
Well, that update can take time so I would assume it's better to workaround it on FO side...

**Comment by me0wing-katt0** [comment]
so this'll be getting a fix? LETS GOOOO

**Comment by me0wing-katt0** [comment]
literally the only reason my performance sucks so much

**Comment by me0wing-katt0** [comment]
the drops are usually way lower with pregenerated chunks, so this is mostly a slow chunk loading issue

**Comment by me0wing-katt0** [comment]
though it could also be mipmaps

**Comment by Madis0** [comment]
> there is usually less drops with pregenerated chunks, so this is mostly a slow chunk loading issue

In that case you could try #305

**Comment by me0wing-katt0** [comment]
sure, is there a fix for this on serverside though? elytra users use a lot of ram

**Comment by me0wing-katt0** [comment]
the only solution i know is pregenerating chunks, but its a bad idea to do this before new 1.19 update because deep dark

**Comment by me0wing-katt0** [comment]
oh right i can just loot the elytras in a 5k-10k radius

**Comment by Madis0** [comment]
> sure, is there a fix for this on serverside though? elytra users use a lot of ram

FO has never been supported on the server. [See the wiki for more info and what can you use on the server.](https://fabulously-optimized.gitbook.io/modpack/readme/server-setup)

**Comment by Madis0** [comment]
Presumably fixed in 3.9.0
